src/util/hooks.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import * as React from 'react'
+
+type SetState<S> = (value: S | ((state: S) => S)) => void
+
+type UseCallback = <T>(callback: T, deps: any[]) => T
```

Comment:
Isn't callback suppose to be a function? same for the return type?

Reply:
Yeah, it should be a function. Whatever function you pass, will get returned out unchanged (but memoized so it's `===` if no deps have changed). Thus, the `T` you pass is the `T` you get (and `T` is a function type). I could try adding a constraint that `<T: Function>`, but these constraints don't really work well in Flow (it seems to randomly ignore them).

Reply:
I have added the type constraint. Now it has to be a function.